pkg/liqo-controller-manager/networking/external-network/wireguard/wggatewayclient_controller.go

```diff
@@ -197,32 +230,47 @@ func (r *WgGatewayClientReconciler) mutateFnWgClientDeployment(deployment *appsv
 	// Forge spec
 	deployment.Spec = wgClient.Spec.Deployment.Spec
 
+	if wgClient.Status.SecretRef != nil {
```

Comment:
With this logic, doesn't a secret with the proper labels take over the secret configured in the spec? Shoudn't the latter have the priority?